src/backend/commands/createas.c

```diff
@@ -1272,6 +1306,9 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *ctx, int
 					ereport(ERROR,
 							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 							 errmsg("system column is not supported on incrementally maintainable materialized view")));
+				/* if subquery reffer vars of  parent tables */
```

Comment:
If EXISTS subquery refers to vars of the upper query, collect these vars.

Reply:
Thank you for recommendation. I will fix it.